src/test/test_bitcoin.cpp

```diff
@@ -30,20 +30,30 @@ ZCJoinSplit *pzcashParams;
 extern bool fPrintToConsole;
 extern void noui_connect();
 
+JoinSplitTestingSetup::JoinSplitTestingSetup()
+{
+    boost::filesystem::path pk_path = ZC_GetParamsDir() / "sprout-proving.key";
+    boost::filesystem::path vk_path = ZC_GetParamsDir() / "sprout-verifying.key";
+    pzcashParams = ZCJoinSplit::Prepared(vk_path.string(), pk_path.string());
+}
+
+JoinSplitTestingSetup::~JoinSplitTestingSetup()
+{
+    delete pzcashParams;
+}
+
 BasicTestingSetup::BasicTestingSetup()
 {
-        assert(init_and_check_sodium() != -1);
-        ECC_Start();
-        pzcashParams = ZCJoinSplit::Unopened();
```

Comment:
This is the cause of the crash. Previously `TestingSetup` had access to `pzcashParams` because it inherits from `BasicTestingSetup`. With this change, it doesn't have that defined any more, leading to a null pointer error in `rpc_wallet_tests` (which uses `TestingSetup`). I think the best solution is to have `TestingSetup` inherit from `JoinSplitTestingSetup`.

Reply:
Oh, and if we do this, the test that was triggering this bug then fails:

```
test/rpc_wallet_tests.cpp(1377): error: in "rpc_wallet_tests/rpc_z_shieldcoinbase_internals": check string(e.what()).find("JoinSplit verifying key not loaded")!= string::npos has failed
```